build-system/babel-plugins/babel-plugin-transform-jss/index.js

```diff
@@ -0,0 +1,139 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Takes a .jss.js file and transforms the `useStyles` export to remove side effects
+ * and directly return the classes map.
+ *
+ * @example
+ * In:
+ * ```
+ * import {createUseStyles} from 'react-jss'
+ *
+ * const JSS = { button: { fontSize: 12 }}
+ * export const CSS = ''
+ * export const useStyles = createUseStyles(JSS);
+ * ```
+ *
+ * Out:
+ * ```
+ * const JSS = { button: { fontSize: 12 }}
+ * const classes = {button: 'button-1'}
+ * export const useStyles = () => classes;
+ * export const CSS = `button-1 { fontSize: 12 }`
+ * ```
+ */
+
+const {create} = require('jss');
+const {default: preset} = require('jss-preset-default');
+
+module.exports = function ({types: t}) {
+  function isJssFile(filename) {
+    return filename.endsWith('.jss.js');
+  }
+
+  function compileJss(JSS) {
+    const jss = create();
+    jss.setup(preset());
+    return jss.createStyleSheet(JSS);
+  }
+
+  const isIdent = (path, ident) => path.node.id && path.node.id.name === ident;
+  const sheetMap = new WeakMap();
+
+  function findAndCompileJss(path, state) {
+    if (sheetMap.has(state.file)) {
+      return;
+    }
+
+    //TODO: @jridgewell, is there a better way to find module scope?
+    const moduleScopePath = path.findParent((p) => p.parent.type === 'File');
+    // Try to find the createUseStyles. The first arg will be the JSS.
+    // Compile the JSS and place into the sheetMap.
+    moduleScopePath.traverse({
+      CallExpression(path) {
+        if (path.node.callee.name !== 'createUseStyles') {
+          return;
+        }
+
+        const {confident, value: JSS} = path.get('arguments')[0].evaluate();
+        if (!confident) {
+          throw new Error(
+            `First argument to createUseStyles must be statically evaluatable.`
+          );
+        }
+        sheetMap.set(state.file, compileJss(JSS));
+      },
+    });
+    if (!sheetMap.has(state.file)) {
+      throw new Error(`Could not find createUseStyles.`);
+    }
+  }
+
+  return {
+    visitor: {
+      VariableDeclarator(path, state) {
+        // TODO: Can I skip the whole file if not jss?
+        const {filename} = state.file.opts;
+        if (!isJssFile(filename)) {
+          return;
+        }
+        findAndCompileJss(path, state);
```

Comment:
Do we really need to process this in 3 separate steps? I think we should just be looking for a call to `createUseStyles`, then injecting the `CSS` directly? Something like https://astexplorer.net/#/gist/fae74275b8270dc35bdb3eb92a9eaaec/dc9015106e5e0ae9de8578bb7f53da3669efc289

Reply:
Nice! We still need to get rid of the `react-jss` import, but the rest of that is possible

Reply:
Another issue: attaching `CSS` to the classes object leaks too much. It could hypothetically also overwrite a real classname